android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/UploadManager.java

```diff
@@ -77,8 +75,8 @@
     private final OkHttpClient okHttpClient = new OkHttpClient.Builder()
             .connectTimeout(30, TimeUnit.SECONDS)
             .readTimeout(30, TimeUnit.SECONDS)
-            .writeTimeout(30, TimeUnit.SECONDS)
-            .retryOnConnectionFailure(false).build();
+            .writeTimeout(15, TimeUnit.MINUTES)
```

Comment:
I'd prefer this defaults to 30 seconds as it was, and then in CRF, you set it to be 15 minutes.